### DISCUSSION TITLE
Unable to build with visual studio, dotnet build works.

### DISCUSSION THREAD

**Discussion by YashC** [dotnet/runtime#75571] [Q&A]
Building with 'dotnet build' using [quick start](https://github.com/dotnet/runtime/blob/main/docs/workflow/building/libraries/README.md#quick-start) instructions works fine but running build or testing in Visual Studio fails with following error. I'm currently using release/6.0 branch for testing. 

> Error	MSB4018	The "ChooseBestP2PTargetFrameworkTask" task failed unexpectedly.
> System.IO.FileNotFoundException: Could not load file or assembly 'Newtonsoft.Json, Version=9.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed' or one of its dependencies. The system cannot find the file specified.
> File name: 'Newtonsoft.Json, Version=9.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'
>    at NuGet.RuntimeModel.JsonRuntimeFormat.ReadRuntimeGraph(TextReader textReader)
>    at NuGet.RuntimeModel.JsonRuntimeFormat.ReadRuntimeGraph(Stream stream)
>    at NuGet.RuntimeModel.JsonRuntimeFormat.ReadRuntimeGraph(String filePath)
>    at Microsoft.DotNet.Build.Tasks.TargetFramework.Sdk.TargetFrameworkResolver..ctor(String runtimeGraph) in /_/src/Microsoft.DotNet.Build.Tasks.TargetFramework.Sdk/src/TargetFrameworkResolver.cs:line 24
>    at Microsoft.DotNet.Build.Tasks.TargetFramework.Sdk.ChooseBestP2PTargetFrameworkTask.Execute() in /_/src/Microsoft.DotNet.Build.Tasks.TargetFramework.Sdk/src/ChooseBestP2PTargetFrameworkTask.cs:line 27
>    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute()
>    at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__26.MoveNext()
> 
> WRN: Assembly binding logging is turned OFF.
> To enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog] (DWORD) to 1.
> Note: There is some performance penalty associated with assembly bind failure logging.
> To turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog].
> 	System.Net.Security.Unit.Tests	C:\Users\***\.nuget\packages\microsoft.dotnet.build.tasks.targetframework.sdk\6.0.0-beta.22457.3\build\Microsoft.DotNet.Build.Tasks.TargetFramework.Sdk.targets	84	

Any suggestions or tips would be helpful to resolve this issue?

**Comment by cmilanschmidt** [comment]
@YashC I have also run into this exact same issue. Did you happen to figure out how to fix it?

**Reply by cmilanschmidt** [reply]
In my case I was building from the v6.0.10 tag. 

I managed to work around this issue by manually copying the `Newtonsoft.Json.dll` from Newtonsoft.Json v9.0.1 (downloaded via NuGet) into `~\.nuget\packages\microsoft.dotnet.build.tasks.targetframework.sdk\6.0.0-beta.22463.7\tools\net472` and `~\.nuget\packages\microsoft.dotnet.build.tasks.targetframework.sdk\6.0.0-beta.22463.7\tools\netcoreapp3.1` directories. I renamed the original `Newtonsoft.Json.dll` files to `Newtonsoft.Json.dll.old` just for the sake of keeping them around.

Once I closed and reopened Visual Studio I was able to build the Microsoft.Extensions.Http library successfully.

**Reply by cmilanschmidt** [reply]
This certainly isn't the _right_ way to resolve this issue, but it got me moving again!
